### DISCUSSION TITLE
BootStrap Tag 공부

### DISCUSSION THREAD

**Discussion by hyk339** [pyunji/Hyundai-ITnE-Team_QuadCore#65] [Q&A]
BootStrap Tag 공부한 내용을 올려주세요!

**Comment by hyk339** [comment]
https://getbootstrap.com/docs/4.6/utilities/flex/
부스트랩 flex 관련된 부분

**Reply by hyk339** [reply]
https://getbootstrap.com/docs/4.6/utilities/sizing/
Sizing(vw, vh, w, h)

**Reply by hyk339** [reply]
https://webclub.tistory.com/356
![image](https://user-images.githubusercontent.com/82932127/130976167-89cd8685-bf24-41ca-b37e-72232876759e.png)

**Comment by chaeyeon4u** [comment]
[navbar BootStrap!](https://getbootstrap.com/docs/4.6/components/navbar/)

**Comment by pyunji** [comment]
- `<meta name="viewport" content="width=device-width, initial-scale=1.0" />` :  웹페이지의 뷰포트 너비를 디바이스 어비에 맞추고 초기화면 배율을 1로 지정 -> 기기에 맞게 대응하는 반응형 웹사이트 제작 가능
	- `viewport`: 웹페이지가 사용자에게 보여지는 영역
- `<div class="d-flex flex-column vh-100">`: 

  - `d-flex`: display가 flexbox container를 생성할 수 있게 적용
  - `flex-column`: 수직 방향으로 세팅
  - `vh-100`: viewport의 height를 100% 차지하게
- `<nav class="navbar navbar-expand-sm bg-dark navbar-dark text-white font-weight-bold justify-content-between">`

  - `navbar`: pc 화면에서는 왼쪽에서 오른쪽으로 길게 늘여진 메뉴바를 뜻하고, 모바일에서는 collapse타입의 메뉴이다(수직정렬)
  - `navbar-expand{-sm|-md|-lg|-xl}"` : navbar가 정해진 breakpoint 까지는 수평정렬되고, 그보다 클 경우 수직정렬 된다.
    - `navbar-expand-sm`: 575픽셀 이하까지는 메뉴바가 수직정렬, 576 이상부터는 수직정렬 된다.

  - `bg-dark` : back-ground 색을 dark로
  - `navbar-dark`: 메뉴바의 배경색을 dark로

  - `justify-content{-정렬기준}`: 
    - start: 왼쪽 정렬
    - end: 오른쪽 정렬
    - center: 가운데 정렬
    - between: 양쪽 요소를 좌우에 붙이고 안쪽 요소를 균일한 간격으로 배치
    - around: 전체 요소를 균일한 간격으로 배치 (양쪽 여백 생김)
- `<a class="navbar-brand" href="./"> `

  - `navbar-brand`: 회사명이나 제품명, 프로젝트 명을 넣음
- `<img src="/resources/images/logo-spring.png" width="30" height="30" class="d-inline-block align-top">`

  - `d-inline-block`: inline요소처럼 배치하지만 block 요소처럼 크기를 가질 수 있음
  - ` align-top`: 수직방향 기준 윗쪽으로 정렬
- `<div class="flex-grow-1 container-fluid">`

  - `flex-grow-1`: 사용가능한 공간을 flex-grow 클래스가 지정된 요소들끼리 숫자에 비례해서 꽉차게 나눠가짐
  - `container-fluid`: 디바이스의 너비에 상관없이 항상 width를 100%로 설정
- `<div class="row h-100">`
  - `row`: 해당 요소의 방향을 row 방향으로 함
  - `h-100`: height를 100% 차지함
- `<div class="col-md-4 p-3 bg-dark">`
  - `col-md-4`: breakpoint인 768px 이상부터는 한 행을 수직으로 12개로 나눈 열 공간 중 4만큼 차지하겠다.
  - `p-3`: padding을 3rem만큼 주겠다.
- `<div class="flex-grow-1" style="height: 0px; overflow-y: auto; overflow-x: hidden;">`

  - `overflow`: 요소에 주어진 공간보다 내용이 더 길 때 어떻게 보일지 결정하는 속성

  - `overflow-y` y축 기준으로 길 때

  - `overflow-x` x축 기준으로 길 때

    - visible (기본값) : 특정 요소가 박스를 넘어 가더라도, 그대로 보여준다.

    - hidden : 부모요소의 범위를 넘어가는 자식 요소의 부분은 보이지 않도록 처리한다. (가로 스크롤바가 나타나지 않을 뿐 브라우저에 따라 세로 스크롤바는 나타남)
    - scroll : 부모요소의 범위를 넘어가는 자식요소의 부분은 보이지 않지만, 사용자가 확인 할 수 있도록 스크롤바를 표시한다. (가로 스크롤바 항상 표시)
    - auto : 부모요소의 범위를 넘어가는 자식요소의 부분이 있을 경우 해당 부분을 보이지 않도록 처리하고, 사용자가 해당 부분을 확인 할 수 있도록 스크롤바를 표시 한다. (내용이 넘칠때만 가로 스크롤바 표시)
- `<%@ include file="/WEB-INF/views/common/menu.jsp" %>` : jsp에서 특정 파일의 내용을 해당 위치에 삽입하게 하는 jsp 문법
- `<ul class="nav flex-column">`: 세로 방향의 내비게이션 메뉴를 명시
- `<li class="nav-item mb-2">`
  - `nav-item`: nav의 항목을 추가할 때 li 태그에 붙여줌
  - `mb-2`: bottom의 margin을 2rem 줌

**Reply by hyk339** [reply]
와..감사합니다!!

**Reply by chaeyeon4u** [reply]
👍 감사합니다!!
